azkaban-common/src/main/java/azkaban/server/SlaRequestUtils.java

```diff
@@ -85,8 +93,42 @@ public static List<SlaOption> parseSlaOptions(final HttpServletRequest req, fina
     return slaOptions;
   }
 
-  private static SlaOption parseSlaSetting(final String set, final String flowName, final List<String> emails,
-      final Map<String, Map<String, String>> alertersConfigs) throws ServletException {
+  private static Map<String, Map<String, List<String>>> getStringListMapFromStringMap(
+      final Map<String, Map<String, String>> alertersConfsFromReqParams) {
+    final Map<String, Map<String, List<String>>> result = new HashMap<>();
+    for (final String alerter : alertersConfsFromReqParams.keySet()) {
+      final Map<String, String> alerterConfigs = alertersConfsFromReqParams.get(alerter);
+      final Map<String, List<String>> processedConfigs = new HashMap<>();
+      for (final String propKey : alerterConfigs.keySet()) {
+        processedConfigs.put(propKey, getListFromString(alerterConfigs.get(propKey),
+            alerter.equals(Emailer.ALERTER_NAME)));
+      }
+      result.put(alerter, processedConfigs);
+    }
+    return result;
+  }
+
+  private static List<String> getListFromString(final String str, final boolean spacesAreDelims) {
+    if (str == null || str.trim().isEmpty()) {
+      return Collections.emptyList();
+    }
+    String delimRegex = "\\s*,\\s*|\\s*;\\s*";
+    if (spacesAreDelims) {
+      delimRegex = "\\s*,\\s*|\\s*;\\s*|\\s+";
+    }
+    return Arrays.asList(str.trim().split(delimRegex));
+  }
+
+  public static Set<String> getSetFromString(final String val) {
+    if (val == null || val.trim().length() == 0) {
```

Comment:
Thoughts on isEmpty?

Reply:
Thanks for pointing this out, that method shouldn't be here. I temporarily copied it from ServerUtils.java file to merge `getSetFromString` and `getListFromString` since both do similar things but I decided to do it in a separate commit. I forgot to delete the method. Deleted now. I'll make sure to use isEmpty() when refactoring :)